### PR TITLE
[SPARK-31320] Fix release script for 3.0.0

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

The release script stops working after https://github.com/apache/spark/commit/d5865493ae71e6369e9f3350dd7e694afcf57298 , as we require `mkdocs 1.0.0`.

This PR upgrades `mkdocs` from 0.1.6.3 to 1.0.4. To do that ruby is also upgraded to 2.5.

This PR also fixes some small issues.

### Why are the changes needed?

to make RC

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

tested by 3.0.0-rc1

### CODE REVIEW

File: dev/create-release/do-release-docker.sh [hunk 1]
```diff
@@ -93,7 +93,7 @@ done
 
 GPG_KEY_FILE="$WORKDIR/gpg.key"
 fcreate_secure "$GPG_KEY_FILE"
-$GPG --export-secret-key --armor "$GPG_KEY" > "$GPG_KEY_FILE"
+$GPG --export-secret-key --armor --pinentry-mode loopback --passphrase "$GPG_PASSPHRASE" "$GPG_KEY" > "$GPG_KEY_FILE"
```
Comment (cloud-fan) [inline dev/create-release/do-release-docker.sh:96]: the modern `gpg` command requires users to input a password, which breaks the workflow of the script. This fixes it.

File: dev/create-release/release-util.sh [hunk 2]
```diff
@@ -159,10 +159,14 @@ function get_release_info {
   export SPARK_PACKAGE_VERSION="$RELEASE_TAG"
 
   # Gather some user information.
-  export ASF_USERNAME=$(read_config "ASF user" "$LOGNAME")
+  if [ -z "$ASF_USERNAME" ]; then
```
Comment (cloud-fan) [inline dev/create-release/release-util.sh:162]: we should respect these env variables if they exist

File: dev/create-release/spark-rm/Dockerfile [hunk 3]
```diff
@@ -33,8 +33,8 @@ ENV DEBCONF_NONINTERACTIVE_SEEN true
 # These arguments are just for reuse and not really meant to be customized.
 ARG APT_INSTALL="apt-get install --no-install-recommends -y"
 
-ARG BASE_PIP_PKGS="setuptools wheel"
-ARG PIP_PKGS="pyopenssl numpy sphinx"
+ARG PIP_PKGS="sphinx==2.3.1 mkdocs==1.0.4 numpy==1.18.1"
+ARG GEM_PKGS="jekyll:4.0.0 jekyll-redirect-from:0.16.0 rouge:3.15.0"
```
Comment (cloud-fan) [inline dev/create-release/spark-rm/Dockerfile:37]: This follows part of https://github.com/apache/spark/pull/27534

**Comment by cloud-fan** [comment]
cc @dongjoon-hyun @HyukjinKwon @nchammas

**Comment by cloud-fan** [comment]
Note that we should forward port this to the master branch. It was done at `branch-3.0` as the policy is to use the script of the corresponding branch to do the RC.

**Review by HyukjinKwon — APPROVED** [review]

File: dev/create-release/spark-rm/Dockerfile [hunk 4]
```diff
@@ -76,10 +75,8 @@ RUN apt-get clean && apt-get update && $APT_INSTALL gnupg ca-certificates && \
   Rscript -e "install.packages(c('curl', 'xml2', 'httr', 'devtools', 'testthat', 'knitr', 'rmarkdown', 'roxygen2', 'e1071', 'survival'), repos='https://cloud.r-project.org/')" && \
   Rscript -e "devtools::install_github('jimhester/lintr')" && \
   # Install tools needed to build the documentation.
-  $APT_INSTALL ruby2.3 ruby2.3-dev mkdocs && \
-  gem install jekyll --no-rdoc --no-ri -v 3.8.6 && \
-  gem install jekyll-redirect-from -v 0.15.0 && \
-  gem install rouge
+  $APT_INSTALL ruby2.5 ruby2.5-dev && \
```
Comment (dongjoon-hyun) [inline dev/create-release/spark-rm/Dockerfile:78]: So, did `ruby2.5` work correctly?

**Comment by dongjoon-hyun** [comment]
cc @rxin and @gatorsmile

File: dev/create-release/spark-rm/Dockerfile (hunk 4 above)
Comment (cloud-fan) [reply dev/create-release/spark-rm/Dockerfile:78]: `mkdocs` requires ruby 2.4+, I googled a bit and seems ubuntu 18 supports ruby 2.5 at most.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Pull request closed without merging**